types/ws/index.d.ts

```diff
@@ -27,6 +27,17 @@ import { SecureContextOptions } from "tls";
 import { URL } from "url";
 import { ZlibOptions } from "zlib";
 
+type FirstArgument<F extends typeof Buffer.from> = F extends (data: infer A) => any ? A : never;
+
+type BufferLike =
+    | string
+    | Buffer
+    | ArrayBuffer
+    | ArrayBufferView
+    | FirstArgument<typeof Buffer.from>
```

Comment:
This is shorter

```suggestion
    | Parameters<BufferConstructor['from']>[0]
```

but Buffer.from has 4 overloads, and Typescript can only choose one, so it returns the last and misses the types from the other 3.